src/librustc_middle/ty/error.rs

```diff
@@ -193,7 +199,8 @@ impl<'tcx> TypeError<'tcx> {
         use self::TypeError::*;
         match self {
             CyclicTy(_) | UnsafetyMismatch(_) | Mismatch | AbiMismatch(_) | FixedArraySize(_)
-            | Sorts(_) | IntMismatch(_) | FloatMismatch(_) | VariadicMismatch(_) => false,
+            | Sorts(_) | IntMismatch(_) | FloatMismatch(_) | VariadicMismatch(_)
+            | TargetFeatureCast(_) => false,
```

Comment:
Well, there is no particular reason. This is my first ~~major~~ non-minor contribution to the compiler, so I basically searched for things similar to what I tried to implement. I wonder if we could merge the two though, since I use the distinction between the two to have an adapted error message.